src/helpers/web.ts

```diff
@@ -120,3 +118,26 @@ export function generateQuery(queryParams: IServiceParams): string {
     .map(([key, value]) => `${snakeCase(key)}=${value}`)
     .join('&')
 }
+
+export function parsePOSTResults(uploadURL: string): {
+  putURL: string
+  resultURL: string
+} {
+  // JS for [[:graph:]] https://www.regular-expressions.info/posixbrackets.html
+  const re = /([\x21-\x7E]+)[\r\n]?/gm
+
+  const matches = uploadURL.match(re)
+
+  if (matches === null) {
+    throw new Error(`Parsing results from POST failed: (${uploadURL})`)
+  }
+
+  if (matches?.length !== 2) {
+    throw new Error(`Incorrect number of urls when parsing results from POST: ${matches.length}`)
```

Comment:
I'd probably add in the full uploadURL in the error

Reply:
In line 131 or line 135? It's not present in line 135, most likely